src/Orchard.Web/Modules/Orchard.OpenId/Module.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Options;
+using Orchard.Environment.Shell;
+using Orchard.Security;
+using Orchard.OpenId.Indexes;
+using Orchard.OpenId.Models;
+using Orchard.Users.Models;
+using Orchard.OpenId.Services;
+using Orchard.FileSystem.AppData;
+using Microsoft.Extensions.Logging;
+using Orchard.Settings;
+
+namespace Orchard.OpenId
+{
+    public class Startup : StartupBase
+    {     
+        private readonly string _certificateFullPath;
+        private const string certificateFileName = "Certificate.pfx";
+        private readonly string tenant;
+
+        public Startup(ShellSettings shellSettings, IAppDataFolder appDataFolder, ILoggerFactory loggerFactory)   
+        {
+            tenant = shellSettings.Name;
+            _certificateFullPath = appDataFolder.Combine(shellSettings.Name, certificateFileName);
+        }
+
+        public override void Configure(IApplicationBuilder builder, IRouteBuilder routes, IServiceProvider serviceProvider)
+        {
+            var baseUrl = serviceProvider.GetRequiredService<ISiteService>().GetSiteSettingsAsync().Result.BaseUrl ?? tenant;
+            builder.UseOpenIddict();
```

Comment:
@pinpointtownes I read an answer from you where you explain Openidict uses by default propietary tokens not meant to be read or verified by a third party. I though this is safer than JWT but less standard and maybe can cause headaches with third party apps not based on .Net.

Am i wrong? Although if I'm right I see interesting to offer both options through Admin UI.
One question: if we use default tokens, do we need to configure sth for each tenant to be sure one token issued by one tenant is not valid for accessing other tenant that issued another token with the same id?